### PR TITLE
Add high-level Python API

### PR DESCRIPTION
**Pull request by martinRenou**
Fix https://github.com/mamba-org/mamba/issues/1645

cc. @DerThorsten 

This is adding two functions `from mamba.api import install, create` with the following signatures:

```python
install(env_name: str, specs: list = [], channels: list = [], target_platform: str = None)
create(env_name: str, specs: list = [], channels: list = [], target_platform: str = None)
```

### CODE REVIEW

File: mamba/mamba/api.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+        if pkg_cache_path is None:
+            # use values from conda
+            pkg_cache_path = self.context.pkgs_dirs
+
+        package_cache = libmambapy.MultiPackageCache(pkg_cache_path)
+        return libmambapy.Transaction(api_solver, package_cache)
+
+
+def install(env_name: str, specs: list = [], channels: list = [], target_platform: str = None):
```
Comment (martinRenou) [inline mamba/mamba/api.py:120]: Not sure what to do if `target_platform` is not provided? Is there a way to know the target platform for which the env was previously created? Also I guess a meaningful default target platform for creating envs should be the current platform?

File: mamba/mamba/api.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    channels : tuple of str
+        The channels from which to pull packages e.g. ['default', 'conda-forge'].
+    Raises
+    ------
+    RuntimeError :
+        If the solver did not find a solution or if the installation failed.
+    """
+    prefix = pathlib.Path(os.environ["CONDA_PREFIX"]) / "envs" / env_name
+    (prefix / "conda-meta").mkdir(parents=True, exist_ok=True)
+    (prefix / "pkgs").mkdir(parents=True, exist_ok=True)
```
Comment (wolfv) [inline mamba/mamba/api.py:135]: Ideally we re-use the one from the "base" prefix here.

Comment (martinRenou) [reply mamba/mamba/api.py:135]: So something like `os.environ["CONDA_PREFIX"]) / "pkgs"`?

Comment (wolfv) [reply mamba/mamba/api.py:135]: Yeah, unfortunately it's not that easy to find the base prefix. 

With micromamba, it's the `$MAMBA_ROOT_PREFIX` environment variable.

With conda it's the `import sys; sys.prefix`, but that assumes that conda is running from the base environment.

Comment (martinRenou) [reply mamba/mamba/api.py:135]: Yeah I was under the impression it wouldn't be easy.

So you're suggesting like:
```python
base_prefix = None
if "MAMBA_ROOT_PREFIX" in os.environ:
    base_prefix = os.environ["MAMBA_ROOT_PREFIX"]
else:
    import sys
    base_prefix = sys.prefix
prefix = base_prefix / "envs" / env_name
(prefix / "conda-meta").mkdir(parents=True, exist_ok=True)
(base_prefix / "pkgs").mkdir(parents=True, exist_ok=True)
```
?

Comment (wolfv) [reply mamba/mamba/api.py:135]: yeah, and maybe make it override-able from a function argument

Comment (martinRenou) [reply mamba/mamba/api.py:135]: thanks!

**Comment by martinRenou** [comment]
Ready for another review :)

**Pull request merged**